### PR TITLE
*: Support information_schema.table_constraints

### PR DESCRIPTION
**Pull request by shenli**
Fix https://github.com/pingcap/tidb/issues/2580
See https://dev.mysql.com/doc/refman/5.7/en/table-constraints-table.html

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @tiancaiamao PTAL

File: executor/aggregate_test.go [hunk 1]
```diff
@@ -269,7 +269,7 @@ func (s *testSuite) TestAggregation(c *C) {
 	result.Check(testkit.Rows("<nil>", "<nil>"))
 
 	result = tk.MustQuery("select count(*) from information_schema.columns")
-	result.Check(testkit.Rows("529"))
+	result.Check(testkit.Rows("535"))
```
Comment (shenli) [inline executor/aggregate_test.go]: TABLE_CONSTRAINTS introduces 6 new columns.

**Comment by tiancaiamao** [comment]
LGTM

File: executor/aggregate_test.go (hunk 1 above)
Comment (ngaut) [inline executor/aggregate_test.go]: Is there another way to avoid hard code the row count?

**Pull request merged**